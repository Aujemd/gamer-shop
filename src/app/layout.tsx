import type { Metadata } from "next";

import { Inter } from "next/font/google";

import "./globals.css";

const inter = Inter({ subsets: ["latin"] });

export const metadata: Metadata = {
    title: "Apply Digital Test",
    description: "Frontend development test for Apply Digital",
};

/**
 * Root layout for the application.
 * This layout wraps all pages and provides a consistent structure.
 * @param props - The component props.
 * @param props.children - The child components to be rendered within this layout.
 * @returns The root layout component.
 */
export default function RootLayout({
    children,
}: Readonly<{
    children: React.ReactNode;
}>) {
    return (
        <html lang="en">
            <body className={inter.className}>{children}</body>
        </html>
    );
}
