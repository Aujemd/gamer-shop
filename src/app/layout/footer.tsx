import type { HTMLAttributes } from "react";

import Link from "next/link";

import { LogoIcon } from "@/icons";

type FooterProps = Readonly<Omit<HTMLAttributes<HTMLElement>, "children">>;

/**
 * Footer component for the application.
 * @param props - The component props.
 * @returns The footer component.
 */
export default function Footer(props: FooterProps) {
    const { className = "", ...rest } = props;
    return (
        <footer {...rest} className={`bg-neutral-700 py-16 flex justify-center items-center ${className}`}>
            <Link
                href="/"
                data-testid="footer-logo-link-test-id"
            >
                <LogoIcon
                    data-testid="footer-logo-icon-test-id"
                />
            </Link>
        </footer>
    );
}
