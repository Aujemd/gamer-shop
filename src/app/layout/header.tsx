import type { HTMLAttributes } from "react";

import Link from "next/link";

import { CartIcon } from "@/icons";

type HeaderProps = Readonly<Omit<HTMLAttributes<HTMLElement>, "children">>;

/**
 * Header component for the application.
 * @param props - The component props.
 * @returns The header component.
 */
export default function Header(props: HeaderProps) {
    const { className = "", ...rest } = props;
    return (
        <header {...rest} className={`bg-surface-secondary  ${className}`}>
            <div className="px-6 py-5 flex items-center justify-between text-cta-fill-primary lg:max-w-desktop mx-auto">
                <h1 className="font-area font-bold text-2xl leading-6 text-center tracking-sm">
                    <Link
                        href="/"
                        data-testid="header-logo-link-test-id"
                    >
                        GamerShop
                    </Link>
                </h1>
                <Link
                    href="/cart"
                    data-testid="header-cart-link-test-id"
                >
                    <CartIcon
                        data-testid="header-cart-icon-test-id"
                    />
                </Link>

            </div>

        </header>
    );
}
