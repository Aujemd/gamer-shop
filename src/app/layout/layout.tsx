import type { PropsWithChildren } from "react";

import { DataFetchingProvider, OrFooter, OrHeader } from "@/shared";

type LayoutProps = Readonly<PropsWithChildren>;

/**
 * Layout component for the application.
 * @param props - The component props.
 * @returns The layout component.
 */
export default function Layout(props: LayoutProps) {
    const { children } = props;
    return (

        <DataFetchingProvider>
            <OrHeader data-testid="layout-or-header-test-id" />
            <main className="min-h-screen">
                {children}
            </main>
            <OrFooter data-testid="layout-or-footer-test-id" />
        </DataFetchingProvider>

    );
}
