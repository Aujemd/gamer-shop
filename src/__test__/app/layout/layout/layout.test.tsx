import { render, screen } from "@testing-library/react";
import React from "react";

import { Layout } from "@/app/layout/index";

describe("layout Component", () => {
    const contentMock = "Hello Layout";

    it("renders correctly", () => {
        const { asFragment } = render(
            <Layout>
                {contentMock}
            </Layout>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders children correctly", () => {
        render(
            <Layout>
                <div data-testid="child-test-id">{contentMock}</div>
            </Layout>,
        );
        const childElement = screen.getByTestId("child-test-id");
        expect(childElement).toBeInTheDocument();
        expect(childElement).toHaveTextContent(contentMock);
    });

    it("renders the Header component", () => {
        render(<Layout>{contentMock}</Layout>);
        const header = screen.getByTestId("layout-or-header-test-id");
        expect(header).toBeInTheDocument();
    });

    it("renders the Footer component", () => {
        render(<Layout>{contentMock}</Layout>);
        const footer = screen.getByTestId("layout-or-footer-test-id");
        expect(footer).toBeInTheDocument();
    });
});
