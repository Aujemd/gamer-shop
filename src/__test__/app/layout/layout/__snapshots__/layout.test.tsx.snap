// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout Component renders correctly 1`] = `
<DocumentFragment>
  <header
    class="bg-surface-secondary "
    data-testid="layout-or-header-test-id"
  >
    <div
      class="px-6 py-5 flex items-center justify-between text-cta-fill-primary lg:max-w-desktop mx-auto xl:px-0"
    >
      <h1
        class="font-area font-bold text-2xl leading-6 text-center tracking-sm"
      >
        <a
          data-testid="or-header-logo-link-test-id"
          href="/"
        >
          GamerShop
        </a>
      </h1>
      <a
        aria-label="View shopping cart"
        data-testid="or-header-cart-link-test-id"
        href="/cart"
      >
        <svg
          data-testid="or-header-cart-icon-test-id"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.9709 15.9883H6.99233C6.72752 15.9883 6.47356 15.8829 6.28631 15.6954C6.09906 15.5079 5.99387 15.2535 5.99387 14.9883C5.99387 14.7231 6.09906 14.4687 6.28631 14.2812C6.47356 14.0936 6.72752 13.9883 6.99233 13.9883H17.4163C18.0839 13.9883 18.7324 13.765 19.2589 13.3537C19.7853 12.9424 20.1594 12.3668 20.3219 11.7183L21.9693 5.22828C22.0067 5.08069 22.01 4.92649 21.9788 4.77745C21.9477 4.62841 21.883 4.48846 21.7896 4.36828C21.6925 4.24501 21.5678 4.14638 21.4255 4.08036C21.2833 4.01434 21.1275 3.9828 20.9709 3.98828H6.7527C6.5467 3.40473 6.16553 2.89928 5.66144 2.54122C5.15735 2.18316 4.55502 1.99002 3.93702 1.98828H2.99847C2.73366 1.98828 2.47969 2.09364 2.29244 2.28117C2.1052 2.46871 2 2.72306 2 2.98828C2 3.2535 2.1052 3.50785 2.29244 3.69539C2.47969 3.88292 2.73366 3.98828 2.99847 3.98828H3.93702C4.16511 3.98163 4.38859 4.05341 4.57026 4.19169C4.75193 4.32998 4.88082 4.52641 4.93549 4.74828L4.9954 5.22828L6.72275 11.9883C5.92832 12.0241 5.18063 12.3745 4.64416 12.9624C4.10769 13.5503 3.82639 14.3276 3.86214 15.1233C3.89789 15.9189 4.24776 16.6678 4.83478 17.2051C5.42181 17.7424 6.1979 18.0241 6.99233 17.9883H7.17206C7.00784 18.4414 6.95508 18.9275 7.01822 19.4054C7.08137 19.8833 7.25857 20.3389 7.53482 20.7336C7.81106 21.1284 8.17823 21.4507 8.60521 21.6731C9.03219 21.8956 9.50642 22.0118 9.98773 22.0118C10.469 22.0118 10.9433 21.8956 11.3703 21.6731C11.7972 21.4507 12.1644 21.1284 12.4406 20.7336C12.7169 20.3389 12.8941 19.8833 12.9572 19.4054C13.0204 18.9275 12.9676 18.4414 12.8034 17.9883H15.1598C14.9956 18.4414 14.9428 18.9275 15.006 19.4054C15.0691 19.8833 15.2463 20.3389 15.5225 20.7336C15.7988 21.1284 16.166 21.4507 16.5929 21.6731C17.0199 21.8956 17.4941 22.0118 17.9755 22.0118C18.4568 22.0118 18.931 21.8956 19.358 21.6731C19.785 21.4507 20.1521 21.1284 20.4284 20.7336C20.7046 20.3389 20.8818 19.8833 20.945 19.4054C21.0081 18.9275 20.9553 18.4414 20.7911 17.9883H20.9709C21.2357 17.9883 21.4896 17.8829 21.6769 17.6954C21.8641 17.5079 21.9693 17.2535 21.9693 16.9883C21.9693 16.7231 21.8641 16.4687 21.6769 16.2812C21.4896 16.0936 21.2357 15.9883 20.9709 15.9883ZM19.6928 5.98828L18.3848 11.2283C18.3302 11.4502 18.2013 11.6466 18.0196 11.7849C17.8379 11.9231 17.6144 11.9949 17.3864 11.9883H8.7696L7.2719 5.98828H19.6928ZM9.98773 19.9883C9.79025 19.9883 9.59721 19.9296 9.43301 19.8198C9.26881 19.7099 9.14084 19.5537 9.06527 19.371C8.9897 19.1882 8.96992 18.9872 9.00845 18.7932C9.04698 18.5992 9.14207 18.421 9.28171 18.2812C9.42135 18.1413 9.59926 18.0461 9.79294 18.0075C9.98662 17.9689 10.1874 17.9887 10.3698 18.0644C10.5523 18.1401 10.7082 18.2683 10.8179 18.4327C10.9276 18.5972 10.9862 18.7905 10.9862 18.9883C10.9862 19.2535 10.881 19.5079 10.6938 19.6954C10.5065 19.8829 10.2525 19.9883 9.98773 19.9883ZM17.9755 19.9883C17.778 19.9883 17.5849 19.9296 17.4207 19.8198C17.2565 19.7099 17.1286 19.5537 17.053 19.371C16.9774 19.1882 16.9577 18.9872 16.9962 18.7932C17.0347 18.5992 17.1298 18.421 17.2694 18.2812C17.4091 18.1413 17.587 18.0461 17.7807 18.0075C17.9744 17.9689 18.1751 17.9887 18.3576 18.0644C18.54 18.1401 18.6959 18.2683 18.8057 18.4327C18.9154 18.5972 18.9739 18.7905 18.9739 18.9883C18.9739 19.2535 18.8687 19.5079 18.6815 19.6954C18.4942 19.8829 18.2403 19.9883 17.9755 19.9883Z"
            fill="#585660"
          />
        </svg>
      </a>
    </div>
  </header>
  <main
    class="min-h-screen"
  >
    Hello Layout
  </main>
  <footer
    class="bg-neutral-700 py-16 flex justify-center items-center "
    data-testid="layout-or-footer-test-id"
  >
    <a
      aria-label="Go to homepage"
      data-testid="or-footer-logo-link-test-id"
      href="/"
    >
      <svg
        data-testid="or-footer-logo-icon-test-id"
        height="45"
        viewBox="0 0 171 45"
        width="171"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M103.048 18.6929H99.5059V25.0439H94.8574V3.87963H103.507C108.547 3.87963 111.612 6.87634 111.612 11.2011V11.2692C111.595 16.1559 107.781 18.6929 103.048 18.6929ZM106.879 11.2863C106.879 9.209 105.431 8.08524 103.099 8.08524H99.5059V14.5554H103.201C105.534 14.5554 106.896 13.1592 106.896 11.3544V11.2863H106.879Z"
          fill="white"
        />
        <path
          d="M124.035 18.6929H120.494V25.0439H115.845V3.87963H124.495C129.535 3.87963 132.6 6.87634 132.6 11.2011V11.2692C132.583 16.1559 128.769 18.6929 124.035 18.6929ZM127.867 11.2863C127.867 9.209 126.419 8.08524 124.087 8.08524H120.494V14.5554H124.189C126.522 14.5554 127.884 13.1592 127.884 11.3544V11.2863H127.867Z"
          fill="white"
        />
        <path
          d="M136.48 25.0269V3.87963H141.123V20.8042H151.648V25.0439H136.48V25.0269Z"
          fill="white"
        />
        <path
          d="M162.242 16.6156V25.0439H157.616V16.7008L149.532 3.87963H154.955L159.954 12.393L165.073 3.89665H170.343L162.242 16.6156Z"
          fill="white"
        />
        <path
          d="M78.7197 2.11646L66.2856 25.0444H71.5371L78.7197 11.7873L85.8684 24.9764L85.9022 25.0444H91.1537L78.7197 2.11646Z"
          fill="white"
        />
        <path
          d="M72.0537 41.7986H68.9309V33.5093H72.0537C74.696 33.5093 76.5148 35.2892 76.5148 37.6455C76.5148 39.9848 74.6789 41.7986 72.0537 41.7986ZM72.0537 34.8485H70.4065V40.4764H72.0537C73.821 40.4764 74.9705 39.3068 74.9705 37.6794C74.9705 36.0351 73.821 34.8485 72.0537 34.8485Z"
          fill="white"
        />
        <path
          d="M88.1552 33.5093V41.7986H86.7443V33.5093H88.1552Z"
          fill="white"
        />
        <path
          d="M102.857 41.975C100.259 41.975 98.5609 40.1096 98.5609 37.6625C98.5609 35.318 100.344 33.3329 102.84 33.3329C104.283 33.3329 105.166 33.7436 106.032 34.4624L105.116 35.5747C104.47 35.0271 103.842 34.6848 102.789 34.6848C101.261 34.6848 100.106 36.0196 100.106 37.6454C100.106 39.3738 101.227 40.6401 102.925 40.6401C103.706 40.6401 104.419 40.3835 104.929 40.007V38.4497H102.789V37.1662H106.321V40.6744C105.506 41.376 104.334 41.975 102.857 41.975Z"
          fill="white"
        />
        <path
          d="M118.314 33.5093V41.7986H116.903V33.5093H118.314Z"
          fill="white"
        />
        <path
          d="M132.992 41.7986H131.52V34.8654H128.896V33.5093H135.598V34.8654H132.975V41.7986H132.992Z"
          fill="white"
        />
        <path
          d="M162.759 33.5093H164.218V40.4737H168.579V41.7986H162.759V33.5093Z"
          fill="white"
        />
        <path
          d="M148.121 36.2797L151.11 41.7986H152.706L148.121 33.3329L143.535 41.7986H145.131L148.121 36.2797Z"
          fill="white"
        />
        <path
          d="M26.0812 0L0.5 44.0922H51.647L26.0812 0ZM16.1863 30.8692H28.1562L22.179 20.5589L26.0812 13.8241L39.8784 37.604H12.2841L16.1863 30.8692Z"
          fill="white"
        />
      </svg>
    </a>
  </footer>
</DocumentFragment>
`;
